core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -108,21 +104,24 @@
 
   private synchronized void appendCurrentBuffer(byte[] buf, int offset, int length)
       throws IOException {
-    if (mAvailableBytes < length) {
-      long bytesRequested = mTachyonFS.requestSpace(mBlockId, length - mAvailableBytes);
-      if (bytesRequested + mAvailableBytes >= length) {
-        mAvailableBytes += bytesRequested;
-      } else {
-        mCanWrite = false;
-        throw new IOException(String.format("No enough space on local worker: fileId(%d)"
-            + " blockId(%d) requestSize(%d)", mFile.mFileId, mBlockId, length - mAvailableBytes));
+    initializeLocalFile();
```

Comment:
The assumption I'm making is if appendCurrentBuffer is being called, this is actually a legitimate request that will cache the block, since we do a lot of cancelRecaches to close the BlockOutStreams for requests that seek/skip for we actually write. Originally I did not want to throw exceptions here because a failure is not necessarily a problem (for example the original problem), but I can see it making sense since initialize is being moved to append. Would you prefer if initializeLocalFile threw IOExceptions? Also, which close method are you referring to?